### ISSUE TITLE
Sensors stopped updating at night

### ISSUE THREAD

**Issue by ErikvO** [home-assistant/iOS#2522]
**iOS device model, version and app version**

Model Name: iPhone 14 Pro Max
Software Version: 7.2.1
App version: 2023.12.1 (2023.508)

**Home Assistant Core Version**
2024.1.2

**Describe the bug**
The iOS app no longer updates the sensor information during the night. I can therefore no longer turn off my charger when the battery level reaches 80%. It used to work flawlessly until recently, presumably when the app was updated.

**To Reproduce**
Charge the phone over night. Then in the morning open the app and look at the battery level graph in Home Assistant. The graph stays at the same level (e.g. 40%) until the app was opened, in which case it suddenly rises to 100%.

**Expected behavior**
Every once in a while the battery level should update. It looks like the app only reports 5% intervals during waking hours, so that is what I expect at night (during charging?) as well.

**Screenshots**
![Battery reporting issue](https://github.com/home-assistant/iOS/assets/11665846/ed1ed3b3-e1de-4084-9d23-4b230e26b577)

**Additional context**
The app was not force closed beforehand. It was left open before going to sleep.

**Comment by ncodee** [comment]
+1, also the geolocation tracker has stopped working for me.

**Comment by bgoncal** [comment]
Can you check if you have Local Push enabled? Otherwise if it works during the day but not at night, iOS my have defined that upgrading during the night is not needed since you dont open the app

**Comment by bgoncal** [comment]
@ErikvO btw iOS 17 now has the option to limit your charge to 80%, you don't need HA for that anymore
![IMG_A53B710A18A8-1](https://github.com/home-assistant/iOS/assets/5808343/78ae63dd-66ba-4ddc-afba-511e8d236554)

**Comment by ErikvO** [comment]
@bgoncal Thank you for your reply and your tip for not needing HA for 80% charging anymore.

Unfortunately I cannot use the tip (yet?), as this feature was only introduced for the latest iPhone 15 models. My settings (in iOS 17.2.1) still show the old `Optimized Battery Charging` toggle. I hope they will add that option to older iPhones in the future, as it should work more reliably than this HA route. As a bonus it would also work when charging in other locations than where I set up the smart-plug.

I checked local push for the 2 phones in the household that need this functionality.
It was enabled for phone 1 and was incorrectly set up for phone 2. It seems phone 1 is behaving normally again, so the issue is resolved there.
I will monitor the situation for phone 2 in the coming days.

**Comment by ErikvO** [comment]
I've checked whether this issue is still present.
Both phones now keep reporting their status at night. 👍

Thinking back, I did change some sensor settings on phone 1. I disabled all sensors except:
- Battery Level
- Battery State
- Focus
- Geocoded Location

Since this change, the Battery level was no longer reported at night. I reverted these changes, and the battery level was still not reported, so I did not think these settings had anything to do with the issue. Maybe they did.

Phone 2 was never set up correctly (and never worked before, as I had just set it up to work like phone 1).

In short: The issue seems to be resolved. Sorry to have wasted your time.

**Comment by bgoncal** [comment]
> I've checked whether this issue is still present.
> 
> Both phones now keep reporting their status at night. 👍
> 
> 
> 
> Thinking back, I did change some sensor settings on phone 1. I disabled all sensors except:
> 
> - Battery Level
> 
> - Battery State
> 
> - Focus
> 
> - Geocoded Location
> 
> 
> 
> Since this change, the Battery level was no longer reported at night. I reverted these changes, and the battery level was still not reported, so I did not think these settings had anything to do with the issue. Maybe they did.
> 
> 
> 
> Phone 2 was never set up correctly (and never worked before, as I had just set it up to work like phone 1).
> 
> 
> 
> In short: The issue seems to be resolved. Sorry to have wasted your time.

No worries, have a nice weekend!
